File: packages/js/core/src/types/UriRedirect.ts
import { Uri } from ".";

import { Tracer } from "@polywrap/tracing-js";

export interface UriRedirect<TUri = string> {
  from: TUri;
  to: TUri;
}

export const sanitizeUriRedirects = Tracer.traceFunc(
  "core: sanitizeUriRedirects",
  (input: UriRedirect<Uri | string>[]): UriRedirect<Uri>[] => {
    const output: UriRedirect<Uri>[] = [];
    for (const definition of input) {
      output.push({
        from: Uri.from(definition.from),
        to: Uri.from(definition.to),
      });
    }

    return output;
  }
);

File: packages/py/core/src/types/uri_redirect.py
from __future__ import annotations
from dataclasses import dataclass
from typing import List

from . import Uri


@dataclass
class UriRedirect:
    from_uri: Uri
    to_uri: Uri


def sanitize_uri_redirects(input: List[UriRedirect]) -> List[UriRedirect]:
    output = []
    for definition in input:
        from_uri = Uri(definition.from_uri)
        to_uri = Uri(definition.to_uri) if isinstance(definition.to_uri, str) else definition.to_uri
        output.append(UriRedirect(from_uri=from_uri, to_uri=to_uri))

    return output
